### PR TITLE
CI upgrades

### PR DESCRIPTION
**Pull request by gaborbernat**
- Travis test with python3.7 and python3.8-dev
- Travis use tox to setup/run tests
- Bump tox default envs from python3.6 to python3.7

### CODE REVIEW

**Comment by gvanrossum** [comment]
This looks great!

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,38 @@
...
   # Specifically request 3.5.1 because we need to be compatible with that.
-  - "3.5.1"
-  - "3.6"
-  - "3.7-dev"
+  - python: 3.5.1
+    dist: trusty
+  - python: 3.6    # 3.6.3  pip  9.0.1
+  - python: 3.7    # 3.7.0  pip 10.0.1
+  - python: 3.8-dev
+  - python: 3.7
```
Comment (ethanhs) [inline .travis.yml]: Might you be able to add names to the lint and self check runs? (see https://docs.travis-ci.com/user/customizing-the-build/#naming-jobs-within-matrices). Having 3 "3.7" environments could be a bit confusing :)

File: .travis.yml [hunk 2]
```diff
@@ -1,21 +1,38 @@
-sudo: false
+if: tag IS present OR type = pull_request OR (branch = master AND type = push)   # we only CI the master, tags and PRs
+
 language: python
 # cache package wheels (1 cache per python version)
 cache: pip
-python:
-  - "3.4"
+dist: xenial
```
Comment (ethanhs) [inline .travis.yml]: While *I* know why xenial is needed for some of these runs, it would be nice to leave a short comment for the future.

**Comment by gaborbernat** [comment]
Done requested changes.

**Comment by gaborbernat** [comment]
But somehow it downloads a wheel on other PRS (^^)

**Comment by ethanhs** [comment]
Sorry let me correct myself. lxml doesn't have wheels for *3.7* on Windows yet. It works otherwise because we are targeting 3.6 currently.

**Comment by gaborbernat** [comment]
actually turns out the latest minor has it so bumped to that

**Comment by gaborbernat** [comment]
@ethanhs now should be all ok :+1:

**Pull request merged**